### DISCUSSION TITLE
Menjalankan code serupa secara berulang di R

### DISCUSSION THREAD

**Discussion by shinddayo** [teamalgoritma/community#232] [Q&A] [answered]
Hallo,
saya memiliki data harian, dimana data tersebut juga diterima secara harian. sehingga data dari hari 1 ke hari lainnya terpisah-pisah file sesuai tanggal harinya.

Untuk melakukan visualisasi dan analysis, maka saya harus read banyak file dari banyak hari.
apakah ada cara jika code untuk read datanya hanya cukup dituliskan sekali namun dijalankan berulang untuk membaca setiap data per harinya.

misal:
data1 <- read.csv("data20220701.csv")
data2 <- read.csv("data20220702.csv")
data3 <- read.csv("data20220703.csv")
.
.
.

data30 <- read.csv("data20220730.csv")

selain itu, pada kondisi di atas dimasukkan dalam banyak objek. bagaimana jika di append/union sehingga jadi 1 dataframe berisi banyak hari

misal:
data <- read.csv("data20220701.csv")
data <- read.csv("data20220702.csv")
data <- read.csv("data20220703.csv")

**Comment by idoali** [comment, marked as answer]
Selamat malam bu

Untuk masalah pertama, ibu bisa menggunakan code berikut : 
```{r}
library(stringr)

the_names <- c("file_1", "file_2")
data_files <- list.files("data/")

for (i in 1:2){
  assign(the_names[i], read.csv(str_c("data/", data_files[i])))
}
```
Yang dilakukan code diatas adalah : 
1. `the_names` adalah vector yang menunjukkan nama object yang akan menyimpan data-data kita nanti. Ingat isinya dalam bentuk `character`
2. Fungsi `list.files` akan memberikan kita list nama-nama file yang terdapat di dalam suatu folder. 
3. Setelah itu kita akan melakukan iterasi dengan fungsi for. Yang akan diiterasi disini adalah object `i`, yang nilainya akan berubah dari 1 sampai 2. 
4. Fungsi `assign(x, y)` akan mengubah karakter `x` menjadi object bernama `x` yang didalamnya sudah terdapat data `y`.  

Untuk masalah kedua, ibu bisa menggunakan code berikut : 
```{r}
library(stringr)

data_files <- list.files("data/")
all_data <- data.frame()

for (i in 1:length(data_names)){
  new_data <- read.csv(str_c("data/", data_files[i]))
  all_data <- rbind(all_data, new_data)
}
```
Yang kita lakukan disini adalah : 
1. Menggunakan fungsi `list.files` seperti pada jawaban dari pertanyaan pertama. 
2. Membuat object bernama `all_data` yang berupa data frame kosong.
3. Melakukan iterasi dengan for. Disini kita akan membuat object `new_data` yang akan selalu diperbarui di setiap iterasi. 
4. Menggabungkan `new_data` dan `all_data` dengan menggunakan fungsi `rbind()`

Berikut jawaban dari kami, semoga bisa membantu bu.

**Reply by shinddayo** [reply]
@idoali it works. Thank you
